loadgen/agent-create-vault.js

```diff
@@ -0,0 +1,118 @@
+import { E } from '@agoric/eventual-send';
+import { Far } from '@agoric/marshal';
+import { amountMath } from '@agoric/ertp';
+// import { allComparable } from '@agoric/same-structure';
+import { multiplyBy, makeRatio } from '@agoric/zoe/src/contractSupport';
+import { pursePetnames, issuerPetnames } from './petnames';
+import { disp } from './display';
+import { allValues } from './allValues';
+
+// This is loaded by the spawner into a new 'spawned' vat on the solo node.
+// The default export function is called with some args.
+
+export default async function startAgent([key, home]) {
+  const { zoe, scratch, agoricNames, wallet } = home;
+
+  console.log(`create-vault: building tools`);
+  const {
+    runBrand,
+    bldBrand,
+    runPurse,
+    bldPurse,
+    treasuryInstance,
+  } = await allValues({
+    runBrand: E(agoricNames).lookup('brand', issuerPetnames.RUN),
+    // bldBrand: E(agoricNames).lookup('brand', issuerPetnames.BLD),
+    bldBrand: E(E(wallet).getIssuer(issuerPetnames.BLD)).getBrand(),
+    runPurse: E(wallet).getPurse(pursePetnames.RUN),
+    bldPurse: E(wallet).getPurse(pursePetnames.BLD),
+    treasuryInstance: E(home.agoricNames).lookup('instance', 'Treasury'),
+  });
+
+  const treasuryPublicFacet = E(zoe).getPublicFacet(treasuryInstance);
+
+  const bldBalance = await E(bldPurse).getCurrentAmount();
+  if (bldBalance.value === BigInt(0)) {
+    throw Error(`create-vault: getCurrentAmount(BLD) broken (says 0)`);
+  }
+  console.log(`create-vault: initial balance: ${disp(bldBalance)} BLD`);
+  const bldToLock = amountMath.make(bldBrand, bldBalance.value / BigInt(100));
+
+  const collaterals = await E(treasuryPublicFacet).getCollaterals();
+  const cdata = collaterals.find(c => c.brand === bldBrand);
+  const priceRate = cdata.marketPrice;
+  const half = makeRatio(BigInt(50), runBrand);
+  const wantedRun = multiplyBy(multiplyBy(bldToLock, priceRate), half);
```

Comment:
Ah, so we're only taking out half the value of the collateral, which gives us 200% collateralization? Can we add a comment to that effect?